### ISSUE TITLE
Differences between Turfjs and Spatialite

### ISSUE THREAD

**Issue by lucacasagrande** [Turfjs/turf-centroid#3]
I noticed that there are some differences between centroids calculated with Turfjs and Spatialite. What do you think about it?

![test_turf](https://cloud.githubusercontent.com/assets/141846/5823525/c89e79c4-a0dd-11e4-8e65-c9c672d2bac1.jpg)

**Comment by morganherlocker** [comment]
relevant GEOS docs:

http://geos.osgeo.org/doxygen/classgeos_1_1algorithm_1_1Centroid.html

> Dimension = 2 - Based on the usual algorithm for calculating the centroid as a weighted sum of the centroids of a decomposition of the area into (possibly overlapping) triangles. The algorithm has been extended to handle holes and multi-polygons. See http://www.faqs.org/faqs/graphics/algorithms-faq/ for further details of the basic approach.
> Dimension = 1 - Computes the average of the midpoints of all line segments weighted by the segment length.
> Dimension = 0 - Compute the average coordinate over all points.

**Comment by morganherlocker** [comment]
Ok @lucacasagrande , so I'm pretty sure I have isolated the difference. Spatiallite uses [geometric decomposition](http://en.wikipedia.org/wiki/Centroid#By_geometric_decomposition) for polygons, then calculates the arithmetic mean on the resulting centroid for the final result. This is kind of an arbitrary choice, so I'm not sure if it matters. For example, all of the turf centroids in the above image look pretty reasonable to me. This would be a performance hit, for no clear win in terms of "precision".

Closing, but totally open to debate here. Thanks for reporting this @lucacasagrande; it's great to get a deeper look at prior art here.
